CONTRIBUTING.md

```diff
@@ -0,0 +1,120 @@
+# Contributing to Merlin
+
+Merlin is a community oriented open-source project and we encourage and value
+any kind of contribution. Thanks for taking the time to contribute 🐫 !
+
+## Code of Conduct
+
+Merlin  adheres to the OCaml Code of Conduct as stated in the [Code of Conduct
+document](CODE_OF_CONDUCT.md). By participating, you are expected to uphold this
+code. Please report unacceptable behavior either to local contacts (listed in
+[here](CODE_OF_CONDUCT.md)) or to someone listed in the upstream [OCaml Code of
+Conduct](CODE_OF_CONDUCT.md).
+
+## Documentation
+
+Merlin (partial and fragmented) documentation can be found in the following
+places:
+- Installation instructions and editor setup can be found in the
+  [README.md](README.md) and the [Github Pages](https://ocaml.github.io/merlin/)
+- Various topics related to editor support and configuration can be found in the
+  [wiki](https://github.com/ocaml/merlin/wiki).
+- The Merlin protocol is described in [this
+  document](https://github.com/ocaml/merlin/blob/master/doc/dev/PROTOCOL.md).
+
+Contribution to the documentation are welcome!
+
+## Question, bug reports and feature requests
+
+We rely on [Github's issue tracker](https://github.com/ocaml/merlin/issues) for
+support questions, feature requests and bug reports.
+
+When reporting an issue, please include a precise reproduction in the bug report
+when that's possible, as it is a very useful tool to investigate. You should
+also check that you are using the latest version of Merlin and that a similar
+issue as not already been submitted.
+
+## Code contributions
+
+ ### Legal Notice
+ When contributing to Merlin, you must agree that you have authored 100% of the
+ content, that you have the necessary rights to the content and that the content
+ you contribute may be provided under the [project's MIT
+ license](https://github.com/ocaml/merlin/blob/master/LICENSE).
+
+
+### Styleguides
+
+As of today, Merlin's codebase does not use a code formatter. When contributing
+code to an existing module, one should adopt the style of the surrounding code.
+Please keep lines under 80 characters.
+
+We plan to move the codebase to ocamlformat in a near future.
+
+Changes unrelated to the issue addressed by a PR should be made in a separate
+PR. Additionally, formatting changes in parts of the code not concerned by a
+specific PR should be proposed in another PR.
+
+### Menhir version
+
+Merlin promotes the generated Menhir parser in its sources. This is done to
+avoid depending on Menhir when installing Merlin. However this also means that
+unnecessary diff will appear when the parser gets re-generated by a different
+version of Menhir. To remove this diff please use version `20201216`:
+```bash
+$ opam pin menhir 20201216
+```
+The generated parser file should only be commited if there is an actual change
+in the grammar.
+
+### Repository organization
+
+
+#### `/src/ocaml`
```

Comment:
I think this section should be explicit about the fact that certain parts of the code (and pointed) should not be modified (because they're vendor-defined).